### PR TITLE
SL510457 proposal: OptiRoute

### PR DESCRIPTION
**Pull request by SL510457**
---
A good PR should include the following parts:

- [ ] A clear title
- [ ] A meaningful message for PR, as well as its commits
- [ ] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [ ] excluding any irrelevant files, such as binaries, text files, or dot files
- [ ] passing tests/CI

### CODE REVIEW

File: project/SL510457/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# OptiRoute: Intelligent Travel Route Planner
+This proposal outlines the workflow and plans for the NYCU NSD term project in Autumn 2023.
+
+## Basic Information
+OptiRoute focuses on solving the Traveling Salesman Problem (TSP), a fundamental challenge in optimization and computer science. At its core, TSP aims to find the most efficient route that visits a set of destinations and returns to the starting point.
```
Comment (yungyuc) [inline project/SL510457/README.md:5]: It's too hard for me to review the long lines.  Please give it a proper break width.

**Review by yungyuc — CHANGES_REQUESTED** [review]
I will continue review once you have it format properly for review.  Please leave a global comment to inform me when it is ready for review.

- [ ] Use proper line width (e.g., 79 characters.)

File: project/SL510457/README.md (hunk 1 above)
Comment (SL510457) [reply project/SL510457/README.md:5]: No problem, I will modify it.

**Comment by yungyuc** [comment]
I saw you "requested a review".  That's not what I asked for.  You did not address my point either.  Please follow the instructions: https://github.com/yungyuc/nsdhw_23au/pull/138#pullrequestreview-1715207127 .

If you are unsure about what I meant, find me on discord or the [meetup this week](https://sciwork.dev/meetup/2023/1108-nycu).

File: project/SL510457/README.md (hunk 1 above)
Comment (yungyuc) [reply project/SL510457/README.md]: You seemed to made some modification but didn't do what I asked for.

Comment (SL510457) [reply project/SL510457/README.md]: I have made some modifications. Is this what you asked for?

File: project/SL510457/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+2. Version control: `git`
+3. Testing framework: `pytest`
+4. Documentation: `README.md`
+
+
+## Schedule
+Planning phase (8 weeks from 10/30 to mm/dd):
+| Week | Date | 內容 |
+|---|---|---|
+| 1 | 10/30 | - Study  for TSP, Christofides algorithm and Haversine Formula       <br> - Start coding the function of `Haversine Function`         <br> - Prepare a presentation skeleton|
```
Comment (yungyuc) [inline project/SL510457/README.md:93]: Table is hard to review so that I skip this part.  You are on your own.

**Review by yungyuc — CHANGES_REQUESTED** [review]
It seems that you modified the line width manually.  It's fine.  But we usually let the editor to do that for us.  Vim, emacs, and all popular editors can do it.

And you did not seem to understand what is a global comment.  What I meant by a global comment is like:

image

These are basic development skills.  I hope you get them sharpen by the end of the semester.

- [ ] Add a link to your project proposal in `project/README.rst`.

**Comment by SL510457** [comment]
Thank you for your feedback. I'm committed to enhancing my skills by the end of the semester!
I have added a link to my project proposal in project/README.rst.

**Pull request merged**